src/helics/core/CoreBroker.cpp

```diff
@@ -888,52 +932,10 @@ void CoreBroker::processCommand(ActionMessage&& command)
                 routeMessage(command);
             }
             break;
-        case CMD_INIT: {
-            auto* brk = getBrokerById(static_cast<GlobalBrokerId>(command.source_id));
-            if (brk != nullptr) {
-                brk->state = connection_state::init_requested;
-            }
-            if (allInitReady()) {
-                if (isRootc) {
-                    LOG_TIMING(global_broker_id_local, "root", "entering initialization mode");
-                    LOG_SUMMARY(global_broker_id_local, "root", generateFederationSummary());
-                    executeInitializationOperations();
-                } else {
-                    LOG_TIMING(global_broker_id_local,
-                               getIdentifier(),
-                               "entering initialization mode");
-                    checkDependencies();
-                    command.source_id = global_broker_id_local;
-                    transmit(parent_route_id, command);
-                }
-            }
-        } break;
-        case CMD_INIT_NOT_READY: {
-            if (allInitReady()) {
-                transmit(parent_route_id, command);
-            }
-            auto* brk = getBrokerById(GlobalBrokerId(command.source_id));
-            if (brk != nullptr) {
-                brk->state = connection_state::connected;
-            }
-        } break;
+        case CMD_INIT:
+        case CMD_INIT_NOT_READY:
         case CMD_INIT_GRANT:
-            if (brokerKey == universalKey) {
-                LOG_SUMMARY(global_broker_id_local,
-                            getIdentifier(),
-                            " Broker started with universal key");
-            }
-            setBrokerState(BrokerState::operating);
-            for (const auto& brk : _brokers) {
-                transmit(brk.route, command);
-            }
-            {
-                timeCoord->enteringExecMode();
-                auto res = timeCoord->checkExecEntry();
-                if (res == MessageProcessingResult::NEXT_STEP) {
-                    enteredExecutionMode = true;
-                }
-            }
+            processInitCommand(command);
```

Comment:
it could, wouldn't make any difference, since CMD_INIT_GRANT doesn't use any elements that could be moved.  The move operation mainly helps when the command contains strings(or raw data).   In this case there is no allocation so move becomes the same as copy or in this case I think it is taking by reference.

Reply:
Okay, so some of the other commands make use of the `SmallBuffer payload` and `std::vector<std::string> stringData` fields in `ActionMessage` but this one doesn't? The function is by reference though, so either way seems like it should be equivalent. There is a branch in `processInitCommand` that uses `std::move`, which seemed a bit strange from a change of ownership perspective.

Reply:
Right,  I think I have been a little lax in the consistency of how the ownership of the action is treated.  In many cases it just reads it so pass by value works well, but in a few paths it needs to resend or pass it on,  In which case it needs to be able to move it even though it just took a reference.    Which would normally indicate use of a unique_ptr but much of the time these things are created on the stack instead of the heap for performance so things end up a bit messy.    Not sure what would be a good way to indicate that the object shouldn't be used after the call.